app/react/Templates/components/FormConfigRelationship.js

```diff
@@ -9,34 +10,55 @@ import { Select } from 'app/ReactReduxForms';
 import { Translate } from 'app/I18N';
 import { Icon } from 'app/UI';
 import PropertyConfigOptions from './PropertyConfigOptions';
-import PropertyConfigOption from './PropertyConfigOption';
 import Tip from '../../Layout/Tip';
 
 export class FormConfigRelationship extends Component {
   static contentValidation() {
     return { required: val => val && val.trim() !== '' };
   }
 
+  constructor(props) {
+    super(props);
+    this.state = { inherit: props.showInheritSelect };
+    this.onInheritChange = this.onInheritChange.bind(this);
+    this.onInheritTypeChange = this.onInheritTypeChange.bind(this);
+  }
+
+  onInheritChange() {
+    const { index } = this.props;
+    if (this.state.inherit) {
+      this.props.resetFormValue(`template.data.properties[${index}].inherit.property`);
+    }
+    this.props.resetFormValue(`template.data.properties[${index}].filter`);
+    this.setState(prevState => ({ inherit: !prevState.inherit }));
+  }
+
+  onInheritTypeChange() {
+    const { index } = this.props;
+    this.props.resetFormValue(`template.data.properties[${index}].filter`);
+  }
+
   render() {
     const {
       index,
       type,
-      inheritPropertyError,
       labelError,
       relationTypeError,
       templates,
       relationTypes,
       templateId,
       showInheritOption,
-      showInheritSelect,
       templateProperties,
       inheritSelectPropertyType,
     } = this.props;
 
     const options = templates.toJS().filter(template => template._id !== templateId);
 
     const labelClass = labelError ? 'form-group has-error' : 'form-group';
-
+    const canBeFilter =
+      !this.state.inherit ||
+      (this.state.inherit &&
+        !['image', 'markdown', 'preview', 'media', 'link'].includes(inheritSelectPropertyType));
```

Comment:
I think it should be `geolocation` instead of `markdown` following the same behavior of regular properties.